### PR TITLE
Forward tests/Makefile targets to run.d ...

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
... to avoid duplication and missmatches between both test runners.

~~Let's see what breaks ...~~

EDIT: Should be fine now

### CODE REVIEW

**Comment by Geod24** [comment]
Instead of `--called-from-make`, why don't you use `.NOTPARALLEL` ? It only works with GNU Make, but so does `-j`, no ?

**Comment by MoonlightSentinel** [comment]
> Instead of `--called-from-make`, why don't you use `.NOTPARALLEL` ?

That could work.

> It only works with GNU Make, but so does `-j`, no ?

Yes, but that makefile isn't compatible with DM Make anyway....

**Comment by MoonlightSentinel** [comment]
Almost finished but semaphore still fails without an apparent reason (except when tests are run sequentially) ...

**Comment by MoonlightSentinel** [comment]
Seems like `gdb` is the culprit on semaphore because it was terminated after starting mutliple `gdb` tests. It seems to pass with  6c62d5d which ensures that mutliple `gdb` tests are mutually exclusive using a dedicated lockfile.

File: test/run.d [hunk 1]
```diff
@@ -89,14 +89,15 @@ int main(string[] args)
...
+    bool runUnitTests, dumpEnvironment;
     int jobs = totalCPUs;
     auto res = getopt(args,
         std.getopt.config.passThrough,
         "j|jobs", "Specifies the number of jobs (commands) to run simultaneously (default: %d)".format(totalCPUs), &jobs,
         "v", "Verbose command output", (cast(bool*) &verbose),
         "f", "Force run (ignore timestamps and always run all tests)", (cast(bool*) &force),
-        "u|unit-tests", "Runs the unit tests", &runUnitTests
+        "u|unit-tests", "Runs the unit tests", &runUnitTests,
+        "e|environment", "Print current environment variables", &dumpEnvironment
```
Comment (Geod24) [inline test/run.d]:
```suggestion
        "e|environment", "Print current environment variables", &dumpEnvironment,
```

For git-friendliness :)

**Review by Geod24 — COMMENTED** [review]
Overall LGTM

File: test/run.d (hunk 1 above)
Comment (MoonlightSentinel) [reply test/run.d]: Done.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**